app/models/ability.rb

```diff
@@ -0,0 +1,17 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    
+    user ||= User.new # guest user (not logged in)
+    
+    if user.has_role? :admin
+      can :manage, :all
+    else
+      can :read, :all
+      can [:create, :destroy], Blog, :user_id => user.id
+      can :update, Blog, :user_id => user.id     #not implemeted in this project.
```

Comment:
Why repeating this statement. You can do this can [:create, :destroy, :update], Blog, :user_id => user.id